2.4/root/usr/share/container-scripts/mongodb/common.sh

```
@@ -134,39 +134,41 @@ function replset_addr() {
 }
 
 # mongo_remove removes the current MongoDB from the cluster
+# optionally takes identifier of replset to connect and address of server to remove
 function mongo_remove() {
   local host
   # if we cannot determine the IP address of the primary, exit without an error
   # to allow callers to proceed with their logic
-  host="$(replset_addr || true)"
+  host="${1:-$(replset_addr || true)}"
+  local mongo_addr
+  mongo_addr="${2:-$(mongo_addr)}"
```

Different than flags to UNIX tools, arguments here are not so convenient to use.
E.g., it is not easy to provide what you want to remove ($2) without providing the replica set address ($1).

Undocumented behavior: is it valid to call `mongo_remove` to remove something that is not an item in the replica set members?

When do we want to specify the host to be something that is not `replset_addr`?

---

> E.g., it is not easy to provide what you want to remove ($2) without providing the replica set address ($1).

@rhcarvalho You can use `""` as a first argument. I know that this is not elegant, but it is possible.

> Undocumented behavior: is it valid to call mongo_remove to remove something that is not an item in the replica set members?

Yes, IMPOV it is valid but should return the appropriate error.

> When do we want to specify the host to be something that is not replset_addr?

#184 uses only hostnames to addressing.

---

> You can use "" as a first argument. I know that this is not elegant, but it is possible.

In this case it would have a different behavior, it would set host to the empty string (and not call `replset_addr`)...



> > When do we want to specify the host to be something that is not replset_addr?
> 
> #184 uses only hostnames to addressing.

And you were arguing there that we should use the replica set addressing :-)
And I agree with you!

That's why I would defer this to be worked on #184, not as a separate PR since it has no clear benefits without that other one (which is work in progress...).

---

> In this case it would have a different behavior, it would set host to the empty string (and not call replset_addr)...

@rhcarvalho Really? `:-` is "If parameter is unset or null"

> And you were arguing there that we should use the replica set addressing :-)
And I agree with you!

Yes i think we should. I am trying to convice @php-coder. But now it is better for me to have more generic function than do not use this function.

> That's why I would defer this to be worked on #184, not as a separate PR since it has no clear benefits without that other one (which is work in progress...).

I agree.
Too late? :-)

---

> Really? :- is "If parameter is unset or null"

Dear God, I should tatoo the contents of `man bash` in my forearms...